bot/exts/backend/branding/_repository.py

```diff
@@ -0,0 +1,240 @@
+        return f"<Event at '{self.path}'>"
+
+
+class BrandingRepository:
+    """
+    Branding repository abstraction.
+
+    This class represents the branding repository's main branch and exposes available events and assets as objects.
+
+    The API is primarily formed by the `get_current_event` function. It performs the necessary amount of validation
+    to ensure that a misconfigured event isn't returned. Such events are simply ignored, and will be substituted
+    with the fallback event, if available.
+
+    Warning logs will inform core developers if a misconfigured event is encountered.
+
+    Colliding events cause no special behaviour - in such cases, the first found active event is returned.
+    We work with the assumption that the branding repository checks for such conflicts and prevents them
+    from reaching the main branch.
+
+    This class keeps no internal state. All `get_current_event` calls will result in GitHub API requests.
+    The caller is therefore responsible for being responsible and caching information to prevent API abuse.
+
+    Requests are made using the HTTP session looked up on the bot instance.
+    """
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+    async def fetch_directory(self, path: str, types: t.Container[str] = ("file", "dir")) -> t.Dict[str, RemoteObject]:
+        """
+        Fetch directory found at `path` in the branding repository.
+
+        The directory will be represented by a mapping from file or sub-directory names to their corresponding
+        instances of `RemoteObject`. Passing a custom `types` value allows only getting files or directories.
+
+        If the request fails, returns an empty dictionary.
+        """
+        full_url = f"{BRANDING_URL}/{path}"
+        log.debug(f"Fetching directory from branding repository: {full_url}")
+
+        async with self.bot.http_session.get(full_url, params=PARAMS, headers=HEADERS) as response:
+            if response.status == 200:
+                json_directory = await response.json()
+            else:
+                log.warning(f"Received non-200 response status: {response.status}")
+                return {}
+
+        return {file["name"]: RemoteObject(file) for file in json_directory if file["type"] in types}
+
+    async def fetch_file(self, download_url: str) -> t.Optional[bytes]:
+        """
+        Fetch file from `download_url`.
+
+        Returns the file as bytes unless the request fails, in which case None is given.
+        """
+        log.debug(f"Fetching file from branding repository: {download_url}")
+
+        async with self.bot.http_session.get(download_url, params=PARAMS, headers=HEADERS) as response:
+            if response.status == 200:
+                return await response.read()
+            else:
+                log.warning(f"Received non-200 response status: {response.status}")
+
+    async def parse_meta_file(self, raw_file: bytes) -> MetaFile:
```

Comment:
Okay, but should it be? It's never used statically.

Reply:
I mentioned this as it doesn’t use class attributes.

Reply:
I know it doesn't; the only instance attribute is the bot instance, which is only there so that we can access the HTTP session.

The question is whether a function should be static every time it theoretically can be. It's never used statically, and never will be, because you need the HTTP session to get the meta file. Decorating it as static adds no benefit apart from making the PyCharm yellow wiggly line go away, and in my opinion communicates the wrong thing.